plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeDynamicFiltering.java

```diff
@@ -0,0 +1,111 @@
+import io.trino.operator.OperatorStats;
+import io.trino.plugin.deltalake.util.DockerizedMinioDataLake;
+import io.trino.spi.QueryId;
+import io.trino.testing.AbstractTestQueryFramework;
+import io.trino.testing.DistributedQueryRunner;
+import io.trino.testing.QueryRunner;
+import org.testng.annotations.Test;
+
+import static com.google.common.base.Verify.verify;
+import static io.airlift.testing.Assertions.assertGreaterThan;
+import static io.trino.SystemSessionProperties.ENABLE_DYNAMIC_FILTERING;
+import static io.trino.plugin.deltalake.DeltaLakeDockerizedMinioDataLake.createDockerizedMinioDataLakeForDeltaLake;
+import static io.trino.plugin.deltalake.DeltaLakeQueryRunner.DELTA_CATALOG;
+import static io.trino.sql.planner.OptimizerConfig.JoinDistributionType.PARTITIONED;
+import static io.trino.tpch.TpchTable.LINE_ITEM;
+import static io.trino.tpch.TpchTable.ORDERS;
+import static java.lang.String.format;
+
+public class TestDeltaLakeDynamicFiltering
+        extends AbstractTestQueryFramework
+{
+    private static final String BUCKET_NAME = "delta-lake-test-dynamic-filtering";
+
+    @Override
+    protected QueryRunner createQueryRunner()
+            throws Exception
+    {
+        verify(new DynamicFilterConfig().isEnableDynamicFiltering(), "this class assumes dynamic filtering is enabled by default");
+        DockerizedMinioDataLake dockerizedMinioDataLake = closeAfterClass(createDockerizedMinioDataLakeForDeltaLake(BUCKET_NAME));
+        QueryRunner queryRunner = DeltaLakeQueryRunner.createS3DeltaLakeQueryRunner(
+                DELTA_CATALOG,
+                "default",
+                // Slowing down the query ensures the dynamic filter has enough time to populate.
+                ImmutableMap.of("delta.max-splits-per-second", "3"),
+                dockerizedMinioDataLake.getMinioAddress(),
+                dockerizedMinioDataLake.getTestingHadoop());
+
+        ImmutableList.of(LINE_ITEM, ORDERS).forEach(table -> {
+            String tableName = table.getTableName();
+            dockerizedMinioDataLake.copyResources("io/trino/plugin/deltalake/testing/resources/databricks/" + tableName, tableName);
+            queryRunner.execute(format("CREATE TABLE %s.%s.%s (dummy int) WITH (location = 's3://%s/%3$s')",
+                    DELTA_CATALOG,
+                    "default",
+                    tableName,
+                    BUCKET_NAME));
+        });
+        return queryRunner;
+    }
+
+    @Test(timeOut = 60_000)
+    public void testDynamicFiltering()
+    {
+        String query = "SELECT * FROM lineitem JOIN orders ON lineitem.orderkey = orders.orderkey AND orders.totalprice > 59995 AND orders.totalprice < 60000";
+        QueryInputStats filteredStats = getQueryInputStats(dynamicFiltering(true), query);
+        QueryInputStats unfilteredStats = getQueryInputStats(dynamicFiltering(false), query);
+        assertGreaterThan(unfilteredStats.numberOfSplits, filteredStats.numberOfSplits);
```

Comment:
We should also assert that results are the same with and without DF (see other usages of assertEqualsIgnoreOrder)